core/embed/rust/src/ui/api/common_c.rs

```diff
@@ -22,7 +22,7 @@ extern "C" fn screen_fatal_error_rust(
     let msg = unsafe { from_c_str(msg) }.unwrap_or("");
     let footer = unsafe { from_c_str(footer) }.unwrap_or("");
 
-    ModelUI::screen_fatal_error(title, msg, footer);
+    fatal_error::error_shutdown(title, msg, footer)
```

Comment:
i'm a little surprised that this compiles. but you should probably annotate `error_shutdown_rust(...) -> !`

Reply:
fixed, see d44b6730f5d5aae946e51fe5c224bd4cbcccaa8e